Library/Homebrew/formula_installer.rb

```diff
@@ -231,6 +231,7 @@ def check_install_sanity
        # TODO: re-enable this on Linux when we merge linuxbrew-core into
        # homebrew-core and have full bottle coverage.
        (OS.mac? || ENV["CI"]) &&
+       !(OS.mac? && OS::Mac.outdated_release?) &&
```

Comment:
Would be nice to instead move this down to the 
```ruby
elsif !Homebrew::EnvConfig.developer? && (!installed_as_dependency? || !formula.any_version_installed?)
```

so instead it's something like
```ruby
elsif !Homebrew::EnvConfig.developer? && (!installed_as_dependency? || !formula.any_version_installed? || !OS.mac? || !OS::Mac.outdated_release?)
```

Reply:
Sounds good to me. Split it up a bit to readability. Though I will say I'm a bit more partial to `!(x && y)` instead of `(!x || !y)`, but happy to stay consistent with surrounding code.